blockchain/internal/spendpruner/db.go

```diff
@@ -76,10 +85,77 @@ func dbUpdateSpendConsumerDeps(dbTx database.Tx, blockHash chainhash.Hash, consu
 	return depsBucket.Put(blockHash[:], serialized)
 }
 
-// dbFetchSpendConsumerDeps uses an existing database transaction to fetch all
-// spend consumer dependency entries in the database.
-func dbFetchSpendConsumerDeps(dbTx database.Tx) (map[chainhash.Hash][]string, error) {
-	depsBucket := dbTx.Metadata().Bucket(spendConsumerDepsBucketName)
+// dbPersistSpendHeights uses an existing database transaction to persist the
+// provided spend heights.
+func dbPersistSpendHeights(dbTx database.Tx, spendHeights map[chainhash.Hash]uint32) error {
+	heightsBucket := dbTx.Metadata().Bucket(spendJournalHeightsBucketName)
+
+	// return immediately if there are no spend heights to persist.
+	if len(spendHeights) == 0 {
+		return nil
+	}
+
+	// Persist all spend height map entries.
+	for blockHash, height := range spendHeights {
+		var b [8]byte
```

Comment:
Why 8 when you're using uint32 which is 4 bytes?

Reply:
Nope, should be 4 bytes there. Updating.